xblock/runtime.py

```diff
@@ -1211,10 +1212,22 @@ class Mixologist(object):
     """
     def __init__(self, mixins):
         """
-        :param mixins: Classes to mixin
+        :param mixins: Classes to mixin or names of classes to mixin.
         :type mixins: `iterable` of `class`
         """
-        self._mixins = tuple(mixins)
+
+        mixin_classes = []
+        for mixin in mixins:
+            if isinstance(mixin, six.text_type):
+                module, cls = mixin.rsplit('.', 1)
+                imported_module = importlib.import_module(module)
+                # Should I catch the attribute error and raise a clearer error on failure?
```

Comment:
I'd say it would be nice to catch errors (including from the rsplit) so that mistakes in the class name would look like "Couldn't import class foobar.mistake.baz: AttributeError: 'module' object has no attribute 'baz'".